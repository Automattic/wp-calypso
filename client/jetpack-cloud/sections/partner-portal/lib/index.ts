export { noop, LICENSE_INFO_MODAL_ID } from './constants';
export { default as getLicenseState } from './get-license-state';
export { default as valueToEnum } from './value-to-enum';
export { publicToInternalLicenseFilter, internalToPublicLicenseFilter } from './license-filters';
export {
	publicToInternalLicenseSortField,
	internalToPublicLicenseSortField,
} from './license-sort-fields';
export { default as ensurePartnerPortalReturnUrl } from './ensure-partner-portal-return-url';
export { default as formatApiPartner } from './format-api-partner';
export { default as getProductTitle } from './get-product-title';
export { default as selectAlphabeticallySortedProductOptions } from './select-alphabetically-sorted-product-options';
export { default as isJetpackBundle } from './is-jetpack-bundle';
export { default as isWooCommerceProduct } from './is-woocommerce-product';
export { default as isWpcomHostingProduct } from './is-wpcom-hosting-product';
export { default as areLicenseKeysAssignableToMultisite } from './are-license-keys-assignable-to-multisite';
