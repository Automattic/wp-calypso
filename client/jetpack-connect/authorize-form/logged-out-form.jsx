// @todo fix this lint error
/* eslint wpcalypso/jsx-classname-namespace: 0 */

/**
 * External dependencies
 */
import React, { Component } from 'react';
import { localize } from 'i18n-calypso';
const debug = require( 'debug' )( 'calypso:jetpack-connect:authorize-form' );

/**
 * Internal dependencies
 */
import addQueryArgs from 'lib/route/add-query-args';
import StepHeader from 'signup/step-header';
import LoggedOutFormLinks from 'components/logged-out-form/links';
import LoggedOutFormLinkItem from 'components/logged-out-form/link-item';
import SignupForm from 'components/signup-form';
import WpcomLoginForm from 'signup/wpcom-login-form';
import config from 'config';
import LocaleSuggestions from 'signup/locale-suggestions';
import versionCompare from 'lib/version-compare';
import HelpButton from '../help-button';
import SiteCard from './site-card';

class LoggedOutForm extends Component {

	componentDidMount() {
		this.props.recordTracksEvent( 'calypso_jpc_signup_view' );
	}

	renderFormHeader() {
		const headerText = this.props.translate( 'Create your account' );
		const subHeaderText = this.props.translate( 'You are moments away from connecting your site.' );
		const { queryObject } = this.props.jetpackConnectAuthorize;
		const siteCard = versionCompare( queryObject.jp_version, '4.0.3', '>' )
			? <SiteCard queryObject={ queryObject } />
			: null;

		return (
			<div>
				<StepHeader
					headerText={ headerText }
					subHeaderText={ subHeaderText } />
				{ siteCard }
			</div>
		);
	}

	submitForm( form, userData ) {
		debug( 'submiting new account', form, userData );
		this.props.createAccount( userData );
	}

	isSubmitting() {
		return this.props.jetpackConnectAuthorize && this.props.jetpackConnectAuthorize.isAuthorizing;
	}

	loginUser() {
		const { queryObject, userData, bearerToken } = this.props.jetpackConnectAuthorize;
		const redirectTo = addQueryArgs( queryObject, window.location.href );
		return (
			<WpcomLoginForm
				log={ userData.username }
				authorization={ 'Bearer ' + bearerToken }
				redirectTo={ redirectTo } />
		);
	}

	renderLocaleSuggestions() {
		if ( ! this.props.locale ) {
			return;
		}

		return (
			<LocaleSuggestions path={ this.props.path } locale={ this.props.locale } />
		);
	}

	renderFooterLink() {
		const { queryObject } = this.props.jetpackConnectAuthorize;
		const redirectTo = addQueryArgs( queryObject, window.location.href );
		const loginUrl = addQueryArgs( { redirect_to: redirectTo }, config( 'login_url' ) );
		return (
			<LoggedOutFormLinks>
				<LoggedOutFormLinkItem href={ loginUrl }>
					{ this.props.translate( 'Already have an account? Sign in' ) }
				</LoggedOutFormLinkItem>
				<HelpButton onClick={ this.clickHelpButton } />
			</LoggedOutFormLinks>
		);
	}

	render() {
		const { userData } = this.props.jetpackConnectAuthorize;
		return (
			<div>
				{ this.renderLocaleSuggestions() }
				{ this.renderFormHeader() }
				<SignupForm
					getRedirectToAfterLoginUrl={ window.location.href }
					disabled={ this.isSubmitting() }
					submitting={ this.isSubmitting() }
					save={ this.save }
					submitForm={ this.submitForm }
					submitButtonText={ this.props.translate( 'Sign Up and Connect Jetpack' ) }
					footerLink={ this.renderFooterLink() }
					suggestedUsername={ userData && userData.username ? userData.username : '' }
				/>
				{ userData && this.loginUser() }
			</div>
		);
	}
}

export default localize( LoggedOutForm );
