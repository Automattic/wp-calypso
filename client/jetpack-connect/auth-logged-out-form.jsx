/**
 * External dependencies
 */
import React, { Component, PropTypes } from 'react';
import debugModule from 'debug';
import { get } from 'lodash';
import { localize } from 'i18n-calypso';

/**
 * Internal dependencies
 */
import HelpButton from './help-button';
import { login } from 'lib/paths';
import LoggedOutFormLinks from 'components/logged-out-form/links';
import LoggedOutFormLinkItem from 'components/logged-out-form/link-item';
import addQueryArgs from 'lib/route/add-query-args';
import LocaleSuggestions from 'components/locale-suggestions';
import SignupForm from 'components/signup-form';
import WpcomLoginForm from 'signup/wpcom-login-form';
import versionCompare from 'lib/version-compare';
import FormattedHeader from 'components/formatted-header';
import SiteCard from './site-card';

const debug = debugModule( 'calypso:jetpack-connect:authorize-form' );

class LoggedOutForm extends Component {
	static propTypes = {
		createAccount: PropTypes.func.isRequired,
		jetpackConnectAuthorize: PropTypes.shape( {
			bearerToken: PropTypes.string,
			isAuthorizing: PropTypes.bool,
			queryObject: PropTypes.object.isRequired,
			userData: PropTypes.object,
		} ).isRequired,
		locale: PropTypes.string,
		path: PropTypes.string,
		translate: PropTypes.func.isRequired,
	};

	componentDidMount() {
		this.props.recordTracksEvent( 'calypso_jpc_signup_view' );
	}

	handleSubmitSignup = ( form, userData ) => {
		debug( 'submiting new account', form, userData );
		this.props.createAccount( userData );
	}

	renderLoginUser() {
		const { queryObject, userData, bearerToken } = this.props.jetpackConnectAuthorize;
		const redirectTo = addQueryArgs( queryObject, window.location.href );
		return (
			<WpcomLoginForm
				log={ userData.username }
				authorization={ 'Bearer ' + bearerToken }
				redirectTo={ redirectTo } />
		);
	}

	renderFormHeader() {
		const { translate } = this.props;
		const headerText = translate( 'Create your account' );
		const subHeaderText = translate( 'You are moments away from connecting your site.' );
		const { queryObject } = this.props.jetpackConnectAuthorize;
		const siteCard = versionCompare( queryObject.jp_version, '4.0.3', '>' )
			? <SiteCard queryObject={ queryObject } />
			: null;

		return (
			<div>
				<FormattedHeader
					headerText={ headerText }
					subHeaderText={ subHeaderText } />
				{ siteCard }
			</div>
		);
	}

	renderLocaleSuggestions() {
		return this.props.locale
			? <LocaleSuggestions path={ this.props.path } locale={ this.props.locale } />
			: null;
	}

	renderFooterLink() {
		const { queryObject } = this.props.jetpackConnectAuthorize;
		const redirectTo = addQueryArgs( queryObject, window.location.href );

		return (
			<LoggedOutFormLinks>
				<LoggedOutFormLinkItem href={ login( { redirectTo } ) }>
					{ this.props.translate( 'Already have an account? Sign in' ) }
				</LoggedOutFormLinkItem>
				<HelpButton onClick={ this.clickHelpButton } />
			</LoggedOutFormLinks>
		);
	}

	render() {
		const {
			isAuthorizing,
			userData,
		} = this.props.jetpackConnectAuthorize;

		return (
			<div>
				{ this.renderLocaleSuggestions() }
				{ this.renderFormHeader() }
				<SignupForm
					getRedirectToAfterLoginUrl={ window.location.href }
					disabled={ isAuthorizing }
					submitting={ isAuthorizing }
					submitForm={ this.handleSubmitSignup }
					submitButtonText={ this.props.translate( 'Sign Up and Connect Jetpack' ) }
					footerLink={ this.renderFooterLink() }
					suggestedUsername={ get( userData, 'username', '' ) }
				/>
				{ userData && this.renderLoginUser() }
			</div>
		);
	}
}

export default localize( LoggedOutForm );
