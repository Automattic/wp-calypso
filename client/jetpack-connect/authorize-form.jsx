/**
 * External dependencies
 */
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { localize } from 'i18n-calypso';
import cookie from 'cookie';

/**
 * Internal dependencies
 */
import Main from 'components/main';
import LoggedOutFormLinks from 'components/logged-out-form/links';
import {
	createAccount,
	authorize,
	goBackToWpAdmin,
	retryAuth,
	goToXmlrpcErrorFallbackUrl
} from 'state/jetpack-connect/actions';
import {
	getAuthorizationData,
	getAuthorizationRemoteSite,
	isCalypsoStartedConnection,
	hasXmlrpcError,
	hasExpiredSecretError,
	isRemoteSiteOnSitesList,
	getAuthAttempts,
	getSiteIdFromQueryObject,
	getUserAlreadyConnected
} from 'state/jetpack-connect/selectors';
import { getCurrentUser } from 'state/current-user/selectors';
import { recordTracksEvent, setTracksAnonymousUserId } from 'state/analytics/actions';
import EmptyContent from 'components/empty-content';
import { requestSites } from 'state/sites/actions';
import { isRequestingSites, isRequestingSite } from 'state/sites/selectors';
import MainWrapper from './main-wrapper';
import HelpButton from './help-button';
import { urlToSlug } from 'lib/url';
import LoggedInForm from './auth-logged-in-form';
import LoggedOutForm from './auth-logged-out-form';

class JetpackConnectAuthorizeForm extends Component {
	static propTypes = {
		authAttempts: PropTypes.number,
		authorize: PropTypes.func,
		calypsoStartedConnection: PropTypes.bool,
		createAccount: PropTypes.func,
		goBackToWpAdmin: PropTypes.func,
		goToXmlrpcErrorFallbackUrl: PropTypes.func,
		isAlreadyOnSitesList: PropTypes.bool,
		isFetchingAuthorizationSite: PropTypes.bool,
		isFetchingSites: PropTypes.bool,
		jetpackConnectAuthorize: PropTypes.shape( {
			queryObject: PropTypes.shape( {
				client_id: PropTypes.string,
				from: PropTypes.string,
			} ).isRequired,
		} ).isRequired,
		recordTracksEvent: PropTypes.func,
		setTracksAnonymousUserId: PropTypes.func,
		requestHasExpiredSecretError: PropTypes.func,
		requestHasXmlrpcError: PropTypes.func,
		requestSites: PropTypes.func,
		retryAuth: PropTypes.func,
		siteSlug: PropTypes.string,
		user: PropTypes.object,
	};

	componentWillMount() {
		// set anonymous ID for cross-system analytics
		const queryObject = this.props.jetpackConnectAuthorize.queryObject;
		if ( queryObject._ui && 'anon' === queryObject._ut ) {
			this.props.setTracksAnonymousUserId( queryObject._ui );
		}
		this.props.recordTracksEvent( 'calypso_jpc_authorize_form_view' );
	}

	isSSO() {
		const cookies = cookie.parse( document.cookie );
		const query = this.props.jetpackConnectAuthorize.queryObject;
		return (
			query.from &&
			'sso' === query.from &&
			cookies.jetpack_sso_approved &&
			query.client_id &&
			query.client_id === cookies.jetpack_sso_approved
		);
	}

	isWCS() {
		return 'woocommerce-services' === this.props.jetpackConnectAuthorize.queryObject.from;
	}

	handleClickHelp = () => {
		this.props.recordTracksEvent( 'calypso_jpc_help_link_click' );
	}

	renderNoQueryArgsError() {
		return (
			<Main className="jetpack-connect__main-error">
				<EmptyContent
					illustration="/calypso/images/illustrations/whoops.svg"
					title={ this.props.translate(
						'Oops, this URL should not be accessed directly'
					) }
					action={ this.props.translate( 'Get back to Jetpack Connect screen' ) }
					actionURL="/jetpack/connect"
				/>
				<LoggedOutFormLinks>
					<HelpButton onClick={ this.handleClickHelp } />
				</LoggedOutFormLinks>
			</Main>
		);
	}

	renderForm() {
		return (
			( this.props.user )
				? <LoggedInForm
					{ ...this.props }
					isSSO={ this.isSSO() }
					isWCS={ this.isWCS() }
				/>
				: <LoggedOutForm
					{ ...this.props }
					isSSO={ this.isSSO() }
					isWCS={ this.isWCS() }
				/>
		);
	}

	render() {
		const { queryObject } = this.props.jetpackConnectAuthorize;

		if ( typeof queryObject === 'undefined' ) {
			return this.renderNoQueryArgsError();
		}

		if ( queryObject && queryObject.already_authorized && ! this.props.isAlreadyOnSitesList ) {
			this.renderForm();
		}

		return (
			<MainWrapper>
				<div className="jetpack-connect__authorize-form">
					{ this.renderForm() }
				</div>
			</MainWrapper>
		);
	}
}

export default connect(
	state => {
		const remoteSiteUrl = getAuthorizationRemoteSite( state );
		const siteSlug = urlToSlug( remoteSiteUrl );
		const requestHasExpiredSecretError = () => hasExpiredSecretError( state );
		const requestHasXmlrpcError = () => hasXmlrpcError( state );
		const siteId = getSiteIdFromQueryObject( state );

		return {
			authAttempts: getAuthAttempts( state, siteSlug ),
			calypsoStartedConnection: isCalypsoStartedConnection( state, remoteSiteUrl ),
			isAlreadyOnSitesList: isRemoteSiteOnSitesList( state ),
			isFetchingAuthorizationSite: isRequestingSite( state, siteId ),
			isFetchingSites: isRequestingSites( state ),
			jetpackConnectAuthorize: getAuthorizationData( state ),
			requestHasExpiredSecretError,
			requestHasXmlrpcError,
			siteSlug,
			user: getCurrentUser( state ),
			userAlreadyConnected: getUserAlreadyConnected( state )
		};
	},
	{
		authorize,
		createAccount,
		goBackToWpAdmin,
		goToXmlrpcErrorFallbackUrl,
		recordTracksEvent,
		setTracksAnonymousUserId,
		requestSites,
		retryAuth,
	}
)( localize( JetpackConnectAuthorizeForm ) );
