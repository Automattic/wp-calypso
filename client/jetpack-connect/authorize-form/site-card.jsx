// @todo fix this lint error
/* eslint wpcalypso/jsx-classname-namespace: 0 */

/**
 * External dependencies
 */
import React from 'react';
import urlModule from 'url';

/**
 * Internal dependencies
 */
import QuerySites from 'components/data/query-sites';
import CompactCard from 'components/card/compact';
import Site from 'blocks/site';
import { decodeEntities } from 'lib/formatting';
import safeImageUrl from 'lib/safe-image-url';

export default function SiteCard( props ) {
	const { site_icon, blogname, home_url, site_url } = props.queryObject;
	const safeIconUrl = site_icon ? safeImageUrl( site_icon ) : false;
	const siteIcon = safeIconUrl ? { img: safeIconUrl } : false;
	const url = decodeEntities( home_url );
	const parsedUrl = urlModule.parse( url );
	const path = ( parsedUrl.path === '/' ) ? '' : parsedUrl.path;
	const site = {
		ID: null,
		url: url,
		admin_url: decodeEntities( site_url + '/wp-admin' ),
		domain: parsedUrl.host + path,
		icon: siteIcon,
		is_vip: false,
		title: decodeEntities( blogname )
	};

	return (
		<CompactCard className="jetpack-connect__site">
		<QuerySites allSites />
		<Site site={ site } />
		</CompactCard>
	);
}
