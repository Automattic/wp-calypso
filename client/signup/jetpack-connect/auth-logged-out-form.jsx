/**
 * External dependencies
 */
import React, { Component } from 'react';
import debugModule from 'debug';
import { localize } from 'i18n-calypso';

/**
 * Internal dependencies
 */
import HelpButton from './help-button';
import { login } from 'lib/paths';
import LoggedOutFormLinks from 'components/logged-out-form/links';
import LoggedOutFormLinkItem from 'components/logged-out-form/link-item';
import addQueryArgs from 'lib/route/add-query-args';
import LocaleSuggestions from 'signup/locale-suggestions';
import SignupForm from 'components/signup-form';
import WpcomLoginForm from 'signup/wpcom-login-form';
import versionCompare from 'lib/version-compare';
import StepHeader from '../step-header';
import SiteCard from './site-card';

const debug = debugModule( 'calypso:jetpack-connect:authorize-form' );

class LoggedOutForm extends Component {
	componentDidMount() {
		this.props.recordTracksEvent( 'calypso_jpc_signup_view' );
	}

	renderFormHeader() {
		const { translate } = this.props;
		const headerText = translate( 'Create your account' );
		const subHeaderText = translate( 'You are moments away from connecting your site.' );
		const { queryObject } = this.props.jetpackConnectAuthorize;
		const siteCard = versionCompare( queryObject.jp_version, '4.0.3', '>' )
			? <SiteCard queryObject={ queryObject } />
			: null;

		return (
			<div>
				<StepHeader
					headerText={ headerText }
					subHeaderText={ subHeaderText } />
				{ siteCard }
			</div>
		);
	}

	submitForm = ( form, userData ) => {
		debug( 'submiting new account', form, userData );
		this.props.createAccount( userData );
	}

	isSubmitting() {
		return this.props.jetpackConnectAuthorize && this.props.jetpackConnectAuthorize.isAuthorizing;
	}

	loginUser() {
		const { queryObject, userData, bearerToken } = this.props.jetpackConnectAuthorize;
		const redirectTo = addQueryArgs( queryObject, window.location.href );
		return (
			<WpcomLoginForm
				log={ userData.username }
				authorization={ 'Bearer ' + bearerToken }
				redirectTo={ redirectTo } />
		);
	}

	renderLocaleSuggestions() {
		if ( ! this.props.locale ) {
			return;
		}

		return (
			<LocaleSuggestions path={ this.props.path } locale={ this.props.locale } />
		);
	}

	renderFooterLink() {
		const { queryObject } = this.props.jetpackConnectAuthorize;
		const redirectTo = addQueryArgs( queryObject, window.location.href );

		return (
			<LoggedOutFormLinks>
				<LoggedOutFormLinkItem href={ login( { redirectTo } ) }>
					{ this.props.translate( 'Already have an account? Sign in' ) }
				</LoggedOutFormLinkItem>
				<HelpButton onClick={ this.clickHelpButton } />
			</LoggedOutFormLinks>
		);
	}

	render() {
		const { userData } = this.props.jetpackConnectAuthorize;
		return (
			<div>
				{ this.renderLocaleSuggestions() }
				{ this.renderFormHeader() }
				<SignupForm
					getRedirectToAfterLoginUrl={ window.location.href }
					disabled={ this.isSubmitting() }
					submitting={ this.isSubmitting() }
					save={ this.save }
					submitForm={ this.submitForm }
					submitButtonText={ this.props.translate( 'Sign Up and Connect Jetpack' ) }
					footerLink={ this.renderFooterLink() }
					suggestedUsername={ userData && userData.username ? userData.username : '' }
				/>
				{ userData && this.loginUser() }
			</div>
		);
	}
}

export default localize( LoggedOutForm );
