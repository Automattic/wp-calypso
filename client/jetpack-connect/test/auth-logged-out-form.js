/**
 * @format
 * @jest-environment jsdom
 */
/**
 * External dependencies
 */
import { shallow } from 'enzyme';
import { identity, noop } from 'lodash';
import React from 'react';

/**
 * Internal dependencies
 */
import LocaleSuggestions from 'components/locale-suggestions';
import { JETPACK_CONNECT_AUTHORIZE_LOGGED_OUT } from './lib/authorize-form';
import { LoggedOutFormTestComponent as LoggedOutForm } from '../auth-logged-out-form';

describe( 'LoggedOutForm', () => {
	test( 'should render', () => {
		const wrapper = shallow(
			<LoggedOutForm
				authorizationData={ JETPACK_CONNECT_AUTHORIZE_LOGGED_OUT }
				createAccount={ noop }
				recordTracksEvent={ noop }
				translate={ identity }
			/>
		);

		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render with locale suggestions', () => {
		const wrapper = shallow(
			<LoggedOutForm
				authorizationData={ {
					...JETPACK_CONNECT_AUTHORIZE_LOGGED_OUT,
					locale: 'es',
				} }
				createAccount={ noop }
				locale="es"
				path="/jetpack/connect/authorize/es"
				recordTracksEvent={ noop }
				translate={ identity }
			/>
		);

		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.find( LocaleSuggestions ) ).toHaveLength( 1 );
	} );
} );
