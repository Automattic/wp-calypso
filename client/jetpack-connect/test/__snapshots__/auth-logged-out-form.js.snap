// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedOutForm should render 1`] = `
<div>
  <Connect(Localized(AuthFormHeader)) />
  <Connect(Localized(SignupForm))
    disabled={false}
    email="email@an.example.site"
    footerLink={
      <LoggedOutFormLinks>
        <LoggedOutFormLinkItem
          href="https://wordpress.com/wp-login.php?redirect_to=https%3A%2F%2Fexample.com%2F%3F_ui%3Djetpack%253AfooBarBaz%26_ut%3Danon%26_wp_nonce%3DfooBarNonce%26blogname%3DExample%2520Blog%26calypso_env%3Dproduction%26client_id%3D98765%26from%3Dbanner-44-slide-1-dashboard%26home_url%3Dhttp%253A%252F%252Fan.example.site%26jp_version%3D5.4%26locale%3Den%26redirect_after_auth%3Dhttp%253A%252F%252Fan.example.site%252Fwp-admin%252Fadmin.php%253Fpage%253Djetpack%26redirect_uri%3Dhttp%253A%252F%252Fan.example.site%252Fwp-admin%252Fadmin.php%253Fpage%253Djetpack%2526action%253Dauthorize%2526_wpnonce%253DfooBarNonce%2526redirect%253Dhttp%25253A%25252F%25252Fan.example.site%25252Fwp-admin%25252Fadmin.php%25253Fpage%25253Djetpack%26scope%3Dadministrator%253AfooBarBaz%26secret%3DfooBarSecret%26site%3Dhttp%253A%252F%252Fan.example.site%26site_icon%3D%26site_lang%3Den_US%26site_url%3Dhttp%253A%252F%252Fan.example.site%26state%3D2%26user_email%3Demail%2540an.example.site%26user_login%3DtheUserLogin&email_address=email%40an.example.site"
        >
          Already have an account? Sign in
        </LoggedOutFormLinkItem>
        <Localized(JetpackConnectHelpButton)
          onClick={[Function]}
        />
      </LoggedOutFormLinks>
    }
    redirectToAfterLoginUrl="https://example.com/?_ui=jetpack%3AfooBarBaz&_ut=anon&_wp_nonce=fooBarNonce&blogname=Example%20Blog&calypso_env=production&client_id=98765&from=banner-44-slide-1-dashboard&home_url=http%3A%2F%2Fan.example.site&jp_version=5.4&locale=en&redirect_after_auth=http%3A%2F%2Fan.example.site%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack&redirect_uri=http%3A%2F%2Fan.example.site%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack%26action%3Dauthorize%26_wpnonce%3DfooBarNonce%26redirect%3Dhttp%253A%252F%252Fan.example.site%252Fwp-admin%252Fadmin.php%253Fpage%253Djetpack&scope=administrator%3AfooBarBaz&secret=fooBarSecret&site=http%3A%2F%2Fan.example.site&site_icon=&site_lang=en_US&site_url=http%3A%2F%2Fan.example.site&state=2&user_email=email%40an.example.site&user_login=theUserLogin"
    submitButtonText="Sign Up and Connect Jetpack"
    submitForm={[Function]}
    submitting={false}
    suggestedUsername=""
  />
</div>
`;

exports[`LoggedOutForm should render with locale suggestions 1`] = `
<div>
  <LocaleSuggestions
    locale="es"
    path="/jetpack/connect/authorize/es"
  />
  <Connect(Localized(AuthFormHeader)) />
  <Connect(Localized(SignupForm))
    disabled={false}
    email="email@an.example.site"
    footerLink={
      <LoggedOutFormLinks>
        <LoggedOutFormLinkItem
          href="https://wordpress.com/wp-login.php?redirect_to=https%3A%2F%2Fexample.com%2F%3F_ui%3Djetpack%253AfooBarBaz%26_ut%3Danon%26_wp_nonce%3DfooBarNonce%26blogname%3DExample%2520Blog%26calypso_env%3Dproduction%26client_id%3D98765%26from%3Dbanner-44-slide-1-dashboard%26home_url%3Dhttp%253A%252F%252Fan.example.site%26jp_version%3D5.4%26locale%3Den%26redirect_after_auth%3Dhttp%253A%252F%252Fan.example.site%252Fwp-admin%252Fadmin.php%253Fpage%253Djetpack%26redirect_uri%3Dhttp%253A%252F%252Fan.example.site%252Fwp-admin%252Fadmin.php%253Fpage%253Djetpack%2526action%253Dauthorize%2526_wpnonce%253DfooBarNonce%2526redirect%253Dhttp%25253A%25252F%25252Fan.example.site%25252Fwp-admin%25252Fadmin.php%25253Fpage%25253Djetpack%26scope%3Dadministrator%253AfooBarBaz%26secret%3DfooBarSecret%26site%3Dhttp%253A%252F%252Fan.example.site%26site_icon%3D%26site_lang%3Den_US%26site_url%3Dhttp%253A%252F%252Fan.example.site%26state%3D2%26user_email%3Demail%2540an.example.site%26user_login%3DtheUserLogin&email_address=email%40an.example.site"
        >
          Already have an account? Sign in
        </LoggedOutFormLinkItem>
        <Localized(JetpackConnectHelpButton)
          onClick={[Function]}
        />
      </LoggedOutFormLinks>
    }
    redirectToAfterLoginUrl="https://example.com/?_ui=jetpack%3AfooBarBaz&_ut=anon&_wp_nonce=fooBarNonce&blogname=Example%20Blog&calypso_env=production&client_id=98765&from=banner-44-slide-1-dashboard&home_url=http%3A%2F%2Fan.example.site&jp_version=5.4&locale=en&redirect_after_auth=http%3A%2F%2Fan.example.site%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack&redirect_uri=http%3A%2F%2Fan.example.site%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack%26action%3Dauthorize%26_wpnonce%3DfooBarNonce%26redirect%3Dhttp%253A%252F%252Fan.example.site%252Fwp-admin%252Fadmin.php%253Fpage%253Djetpack&scope=administrator%3AfooBarBaz&secret=fooBarSecret&site=http%3A%2F%2Fan.example.site&site_icon=&site_lang=en_US&site_url=http%3A%2F%2Fan.example.site&state=2&user_email=email%40an.example.site&user_login=theUserLogin"
    submitButtonText="Sign Up and Connect Jetpack"
    submitForm={[Function]}
    submitting={false}
    suggestedUsername=""
  />
</div>
`;
